optimum/habana/diffusers/pipelines/pipeline_utils.py

```diff
@@ -370,6 +368,16 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
         bf16_full_eval = kwargs.get("torch_dtype", None) == torch.bfloat16
         kwargs["bf16_full_eval"] = bf16_full_eval
 
+        # Need to load custom ops lists before instantiating htcore
+        if isinstance(kwargs["gaudi_config"], str):
```

Comment:
```suggestion
        if isinstance(kwargs.get("gaudi_config"), str):
```

This way, if "gaudi_config" is not a key in kwargs, `kwargs.get("gaudi_config")` will return `None` by default, and `isinstance(None, str)` will return `False`.

Reply:
This is an issue in one of the `test_textual_inversion` test

Reply:
If "gaudi_config" is not a key in kwargs, we should also fix the else condition

Reply:
@yafshar Thanks for catching this.   Actually I accidentally pushed this commit into this PR.  I have a separate PR for this fix so let's switch this discussion to there:  #1655.   I will remove it from this PR (doesn't belong here :)

Reply:
@dsocek I suggested a change in https://github.com/huggingface/optimum-habana/pull/1655 adding it here, solved the issue 

```sh
>>> RUN_SLOW=true GAUDI2_CI=1 python -m pytest tests/test_diffusers.py -s -v -k  test_textual_inversion
================================== 2 passed, 151 deselected
```

Reply:
@yafshar Thanks for great review.  I removed the 2 commits from this PR and updated #1655 with your suggested fix plus added my fix to handle cases where gaudi config is actually GaudiConfig type, like in [this example](https://github.com/huggingface/optimum-habana/blob/main/tests/test_diffusers.py#L690)